cmd/dockerd/daemon.go

```diff
@@ -494,6 +494,14 @@ func initRouter(opts routerOptions) {
 func (cli *DaemonCli) initMiddlewares(s *apiserver.Server, cfg *apiserver.Config, pluginStore plugingetter.PluginGetter) error {
 	v := cfg.Version
 
+	buildkitEnabled := "false"
+	if cli.Config.Buildkit {
+		buildkitEnabled = "true"
+	}
+	specialHeaders := middleware.SpecialHeaders{middleware.BuildkitHeader: buildkitEnabled}
+	special := middleware.NewSpecialMiddleware(specialHeaders)
```

Comment:
Why is this a middleware?

Reply:
I guess it's because all the other things that were setting headers were middlewares as well.

Reply:
@tonistiigi This is how the request headers are set, `Docker-Experimental` for one example. In an early discussion, we wanted to have `_ping` endpoint to carry the information of whether buildkit is enabled on daemon. And this middleware sets the header to `_ping`.

Reply:
This looks very weird that API command that does not have anything to do with builder returns `Buildkit=true`.